libPhoneNumber/Internal/NBRegularExpressionCache.m

```diff
@@ -0,0 +1,60 @@
+//
+//  NBRegularExpressionCache.m
```

Comment:
This part of the change should be broken out to a separate PR because it has nothing to do with the short number support.

Reply:
I can pull this out into its own PR, though that depends on whether we still want to have NBRegExMatcher. If we don't want the separate NBRegExMatcher, I can remove this class as well.